tapdb/universe_stats.go

```diff
@@ -104,22 +109,238 @@ type BatchedUniverseStats interface {
 	BatchedTx[UniverseStatsStore]
 }
 
+// eventQuery is used to query for events within a given time range.
+type eventQuery struct {
+	// startTime is the start time of the query.
+	startTime int64
+
+	// endTime is the end time of the query.
+	endTime int64
+}
+
+// eventQueryBucket is the interval that we'll use to bucket similar queries
+// into.
+const eventQueryBucket = time.Hour
+
+// newEventQuery creates a new event query from the given query.
+func newEventQuery(q universe.GroupedStatsQuery) eventQuery {
+	// For both the start and time time, we'll round down to the nearest
+	// hour. This'll serve to bucket queries into hourly buckets.
+	startTime := q.StartTime.UTC().Truncate(eventQueryBucket).Unix()
```

Comment:
Do we want to implement some minimum query timespan of one hour here? Especially since the caller doesn't explicitly know we're truncating.

IIUC right now start and end times < 1 hr apart would return nothing.

Reply:
> IIUC right now start and end times < 1 hr apart would return nothing.



Hmm, def need to tighten up with some tests. The intention here is that if you have start+end > 1 hr apart, then it's just the end+start that get truncated down. So `start=11:15`, and `end=2:19` goes to `start=11:00`, `end=2:00`.